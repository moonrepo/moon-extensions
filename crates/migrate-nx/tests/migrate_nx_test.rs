use moon_pdk_test_utils::{create_extension, ExecuteExtensionInput};
use starbase_sandbox::{assert_snapshot, create_empty_sandbox, create_sandbox};
use std::fs;

mod migrate_nx {
    use super::*;

    #[test]
    fn converts_root_files() {
        let sandbox = create_sandbox("root");
        let plugin = create_extension("test", sandbox.path());

        plugin.execute_extension(ExecuteExtensionInput {
            args: vec![],
            context: plugin.create_context(sandbox.path()),
        });

        assert!(!sandbox.path().join("nx.json").exists());
        assert!(!sandbox.path().join("workspace.json").exists());
        assert!(sandbox.path().join(".moon/tasks/node.yml").exists());
        assert!(sandbox.path().join(".moon/workspace.yml").exists());

        assert_snapshot!(fs::read_to_string(sandbox.path().join(".moon/tasks/node.yml")).unwrap());
        assert_snapshot!(fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap());
    }

    mod projects {
        use super::*;

        #[test]
        fn uses_defaults() {
            let sandbox = create_empty_sandbox();
            let plugin = create_extension("test", sandbox.path());

            plugin.execute_extension(ExecuteExtensionInput {
                args: vec![],
                context: plugin.create_context(sandbox.path()),
            });

            assert_snapshot!(
                fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap()
            );
        }

        #[test]
        fn inherits_layout() {
            let sandbox = create_empty_sandbox();
            sandbox.create_file(
                "nx.json",
                r#"
{
  "workspaceLayout": {
    "appsDir": "applications",
    "libsDir": "libraries"
  }
}"#,
            );

            let plugin = create_extension("test", sandbox.path());

            plugin.execute_extension(ExecuteExtensionInput {
                args: vec![],
                context: plugin.create_context(sandbox.path()),
            });

            assert_snapshot!(
                fs::read_to_string(sandbox.path().join(".moon/workspace.yml")).unwrap()
            );
        }
    }
}
