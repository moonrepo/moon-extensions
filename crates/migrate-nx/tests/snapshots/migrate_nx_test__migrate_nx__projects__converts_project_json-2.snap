---
source: crates/migrate-nx/tests/migrate_nx_test.rs
expression: "fs::read_to_string(sandbox.path().join(\"baz/moon.yml\")).unwrap()"
---
fileGroups:
  default:
  - '**/*'
  production:
  - '!jest.config.ts'
  - '!**/?(*.)+(spec|test).ts'
language: javascript
platform: node
tasks:
  build:
    command: moon-nx execute @nx/js:tsc
    args:
    - --tsConfig
    - $workspaceRoot/baz/tsconfig.lib.json
    - --main
    - $workspaceRoot/baz/src/main.ts
    deps:
    - ^:build
    inputs:
    - '@group(production)'
    outputs:
    - /dist/baz
  build.production:
    extends: build
    args:
    - --tsConfig
    - $workspaceRoot/baz/tsconfig-prod.lib.json

